src/keyboard/mod.rs

```diff
@@ -124,14 +67,23 @@ impl Display for Key {
 }
 
 impl Key {
-    fn to_key_id(self) -> Result<u8> {
+    fn to_key_id_12(self) -> Result<u8> {
         match self {
             Key::Button(n) if n >= 12 => Err(anyhow!("invalid key index")),
             Key::Button(n) => Ok(n + 1),
             Key::Knob(n, _) if n >= 3 => Err(anyhow!("invalid knob index")),
             Key::Knob(n, action) => Ok(13 + 3*n + (action as u8)),
         }
     }
+
+    fn to_key_id_16(self) -> Result<u8> {
+        match self {
+            Key::Button(n) if n >= 16 => Err(anyhow!("invalid key index")),
+            Key::Button(n) => Ok(n + 1),
+            Key::Knob(n, _) if n >= 3 => Err(anyhow!("invalid knob index")),
+            Key::Knob(n, action) => Ok(16 + 3*n + (action as u8)),
```

Comment:
Shouldn't it be 17 here?

Reply:
That's the thing. I didn't pass `16` as a parameter to the function because it really does seem to start on 16 (at least in my model).

Reply:
Ok, keys >= 16 are rejected, but key 15 is accepted and returns Ok(16). Doesn't it overlap with Knob(0)?

Reply:
This is the only major question left ↑

Reply:
I think the version I just pushed should sort that out.